smart-contracts/ethereum/contracts/libraries/utils/BytesUtils.sol

```diff
@@ -0,0 +1,18 @@
+pragma solidity ^0.4.24;
+
+library BytesUtils {
+
+    uint private constant BYTES_HEADER_SIZE = 32; // Can't actually use constants within in-line assembly but figured it'd be good to define anyways
+
+    function toBytes32(bytes _bytes) external pure returns (bytes32 retBytes32) {
+        if (_bytes.length == 0) {
+            return bytes32(0);
+        }
+
+        bytes memory inBytes = _bytes;
+
+        assembly {
+            retBytes32 := mload(add(inBytes, /* BYTES_HEADER_SIZE */ 32))
+        }
+    }
+}
```

Comment:
Our eternal storage to be as generic as possible, it has a mapping of bytes, except for author tags that have byte32, but they aren't compatible. If you add them, you can't. So you have to convert the bytes32 to a bytes first. And to ensure you can use them in reverse you have to convert back.